lib/nx_compress.c

```diff
@@ -90,14 +90,14 @@ uLong nx_compressBound(uLong sourceLen)
 
 #ifdef ZLIB_API
 
-int compress(Bytef *dest, uLongf *destLen, const Bytef *source, uLong sourceLen)
-{
-	return compress2(dest, destLen, source, sourceLen, Z_DEFAULT_COMPRESSION);
-}
+int compress(Bytef *dest, uLongf *destLen, const Bytef *source,
+	     uLong sourceLen)  __attribute__((alias("compress2")));
 
-int compress2(Bytef *dest, uLongf *destLen, const Bytef *source, uLong sourceLen, int level)
+int compress2(Bytef *dest, uLongf *destLen, const Bytef *source,
+	      uLong sourceLen, int lvl)
 {
 	int rc=0;
+	int level = lvl ? lvl : Z_DEFAULT_COMPRESSION;
```

Comment:
Is this change expected to be here?

Reply:
This is wrong.  `compress()` and `compress2()` have different interfaces.  There is no guarantee the register holding `lvl` will be zero when called from `compress()`.
Unfortunately, we can only alias functions that have the exact same interface.

Reply:
Removed!